File: schemas/attendance.py
from . import ma

class AttendanceSchema(ma.Schema):

    class Meta:
        strict = True

    id = ma.Int(dump_only=True)

    primary = ma.Bool()
    secondary = ma.Bool()
    additional = ma.Str()

    member = ma.Nested('MemberSchema', only=('id', 'ref'), dump_only=True)

    event = ma.Nested('EventSchema', only=('id', 'ref'), dump_only=True)

    ref = ma.URLFor('attendance', id='<id>')

File: src/resources/attendance/attendance.ts
import { BaseResource } from "resources/resource.base";
import { Attendance } from "interfaces/attendance";
import { MemberResource } from "resources/member";
import { Type, Transform } from "class-transformer";
import { EventResource } from "resources/event/event";
import { Allow, ValidateNested } from "class-validator";
import { ApiModelProperty, ApiModelPropertyOptional } from "@nestjs/swagger";
import { RefResource } from "resources/resource.ref";
import { Member } from "interfaces/member";

export class AttendanceResource extends BaseResource implements Attendance {

    @ApiModelProperty()
    @Allow()
    id: number;
    
    @ApiModelProperty()
    @Allow()
    primary: boolean;

    @ApiModelProperty()
    @Allow()
    secondary: boolean;
    
    @ApiModelPropertyOptional()
    @Allow()
    additional?: string;

    @ApiModelPropertyOptional()
    @Allow()
    @Type(() => MemberResource)
    @Transform((m: MemberResource) => {return {id: m.id, ref: m.ref}})
    member: RefResource

    @ApiModelPropertyOptional()
    @Allow()
    @Type(() => EventResource)
    @Transform((e: EventResource) => {return {id: e.id, ref: e.ref}})
    event: RefResource

    get prefix(): string {
        return "/attendance"
    }
    
    getResourceName(): string {
        return "Attendance"
    }
}